internal/dcloud/resource_inbound_proxy.go

```diff
@@ -0,0 +1,144 @@
+package dcloud
+
+import (
+	"context"
+	"github.com/cisco-open/dcloud-tb-go-client/tbclient"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceInboundProxy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceInboundProxyCreate,
+		ReadContext:   resourceInboundProxyRead,
+		DeleteContext: resourceInboundProxyDelete,
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"topology_uid": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"nic_uid": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"nic_ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_vm_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tcp_port": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"url_path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"hyperlink": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ssl": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+			"show_hyperlink": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceInboundProxyCreate(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	c := i.(*tbclient.Client)
+
+	var diags diag.Diagnostics
+
+	inboundProxyRule := tbclient.InboundProxyRule{
+		Topology: &tbclient.Topology{
+			Uid: data.Get("topology_uid").(string),
+		},
+		VmNicTarget: &tbclient.TrafficVmNicTarget{
+			Uid: data.Get("nic_uid").(string),
+			Vm: &tbclient.Vm{
```

Comment:
Shouldn't need to input the VM name

Reply:
Same as above. It game 400 BR with no VM name.